controllers/fluxcd/git-repository-controller.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fluxcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	"kubesphere.io/devops/pkg/api/devops/v1alpha3"
+	"kubesphere.io/devops/pkg/utils/k8sutil"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups=devops.kubesphere.io,resources=gitrepositories,verbs=get;list;watch;update;delete
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;create;update;patch;delete
```

Comment:
Do we really need the `DELETE` permission here?

Reply:
```golang
```

Reply:
Thanks a lot for your review.

And I think the git-repository-controller only needs the **READ** permission(get list watch) on _devops.kubesphere.io/gitrepositories_ and **CRUD** permission on _source.toolkit.fluxcd.io/gitrepositories_ like below:
```golang
//+kubebuilder:rbac:groups=devops.kubesphere.io,resources=gitrepositories,verbs=get;list;watch
//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
//+kubebuilder:rbac:groups="source.toolkit.fluxcd.io",resources=gitrepositories,verbs=get;list;create;update;delete
```
but I actually have a little confused about the kubebuilder RABC marker. 
I run `make manifests` and I got a ClusterRole named manager-role in config/rbac/role.yaml. And the manager-role ClusterRole binding to a single ServiceAccount. The ks-devops-controller mount that ServiceAccount. Is that mean all the controller like jenkins、argocd in the controller-manager own all the permission specified in the role.yaml? but not the permission specified by the kubebuilder RBAC marker above each Reconciler?